### PR TITLE
Frontend/fix: Added events to track ride request And Refctored code for adding events

### PR DESCRIPTION
**Pull request by Rohit4997**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [x] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

**Review by kavya-shree-s — DISMISSED** [review]

File: Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/OverlaySheetService.java [hunk 1]
```diff
@@ -321,7 +308,7 @@ public void onViewHolderBind(SheetAdapter.SheetViewHolder holder, int position,
                             });
                         }
                     } catch (Exception e) {
-                        firebaseLogEventWithParams("exception", "request_button_click", e.toString());
+                        firebaseLogEventWithParams("exception_request_button_click", "request_button_click", Objects.requireNonNull(e.getMessage()).substring(0, 40));
```
Comment (Vignesh-772) [inline Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/OverlaySheetService.java:311]: here also.

File: Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/RideRequestUtils.java [hunk 2]
```diff
@@ -406,4 +417,53 @@ public static void updateDriverStatus(Boolean status, String mode, Context conte
             }
         });
     }
+
+    public static void addRideReceivedEvent(JSONObject entity_payload, Bundle rideRequestBundle, SheetModel model, String event, Context context) {
+        ExecutorManager.runOnBackgroundThread(() -> {
```
Comment (Vignesh-772) [inline Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/RideRequestUtils.java:422]: is it okay to push log for every ride request to clevertap?

File: Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/OverlaySheetService.java [hunk 3]
```diff
@@ -537,7 +487,7 @@ public IBinder onBind(Intent intent) {
                 mediaPlayer.setOnPreparedListener(mp -> mediaPlayer.setWakeMode(getApplicationContext(), PowerManager.PARTIAL_WAKE_LOCK));
             }
         } catch (Exception e) {
-            firebaseLogEventWithParams("exception", "on_bind", e.toString());
+            firebaseLogEventWithParams("exception_in_on_bind", "on_bind", Objects.requireNonNull(e.getMessage()).substring(0, 40));
```
[the remark this answers is not in the document]
Comment (Rohit4997) [reply Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/OverlaySheetService.java:490]: Here e cannot be bull

[the remark this answers is not in the document]
Comment (Vignesh-772) [reply Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/OverlaySheetService.java:490]: then why assertion is required?

File: Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/RideRequestUtils.java (hunk 2 above)
Comment (Rohit4997) [reply Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/RideRequestUtils.java:422]: Yes this is what we want

**Review by Vignesh-772 — APPROVED** [review]

**Pull request merged**